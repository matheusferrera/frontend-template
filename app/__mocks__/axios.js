/* eslint-disable no-undef */
import { loginMockData, registerMockData, userMockData } from "../test/mockData";

// Cria um objeto simulado que atua como uma vers√£o personalizada do Axios
const customAxios = {
  mockLogin: jest.fn().mockResolvedValue(loginMockData),
  mockRegister: jest.fn().mockResolvedValue(registerMockData),
  mockUser: jest.fn().mockResolvedValue(userMockData),
  mockDate: jest.fn().mockResolvedValue(new Date()),
};

export default customAxios;
