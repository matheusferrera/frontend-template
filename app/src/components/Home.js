import React from "react";

// import UserService from "../services/user.service";

const Home = () => {
  // const [content, setContent] = useState("");
  // const [monsters, setMonsters] = useState([]);

  // useEffect(() => {
  //   UserService.getPublicContent().then(
  //     response => {
  //       setContent(response.data);
  //     },
  //     error => {
  //       const _content = (error.response && error.response.data) || error.message || error.toString();

  //       setContent(_content);
  //     },
  //   );
  // }, []);

  // useEffect(() => {
  //   fetch("https://jsonplaceholder.typicode.com/users")
  //     .then(response => response.json())
  //     .then(users => setMonsters(users));
  // }, []);

  // console.log(monsters);
  return (
    <div className="container">
      <header className="jumbotron">
        {/* <h3>{content}</h3> */}
        <h3>Página em construção</h3>
      </header>
      {/* <div className="card-list">
        {monsters.map(monster => {
          return (
            <div
              key={monster.id}
              className="card-container"
            >
              <img
                alt={`monster ${monster.name}`}
                src={`https://robohash.org/${monster.id}?set=set2&size=180x180`}
              />
              <h2>{monster.name}</h2>
              <p>{monster.email}</p>
            </div>
          );
        })}
      </div> */}
    </div>
  );
};

export default Home;
