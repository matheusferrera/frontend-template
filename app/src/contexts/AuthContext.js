import React, { createContext, useContext, useState } from "react";
import authService from "../services/auth.service";
import isEqual from "lodash/isEqual";

const AuthContext = createContext();

/**
 * AuthProvider component that handles user authentication and authorization.
 * @param {object} props - React component props.
 * @param {React.ReactNode} props.children - The child components to render.
 * @returns {React.ReactNode} - The rendered child components wrapped in AuthContext.Provider.
 */
// eslint-disable-next-line react/prop-types
export const AuthProvider = ({ children }) => {
  // State variables to store user and token
  const [user, setUser] = useState(null);
  const [token, setToken] = useState(null);

  /**
   * Function to handle user login.
   * @param {string} email - The user's email.
   * @param {string} password - The user's password.
   * @returns {Promise} - A promise that resolves to the user details.
   */
  const login = (email, password) => {
    return authService
      .login(email, password)
      .then(token => {
        // Ensure that token is truthy before setting it
        if (token) {
          setToken(token);
          return authService.getAuthUser(token);
        } else {
          throw new Error("Invalid token received");
        }
      })
      .then(userDetails => {
        // Ensure that userDetails is truthy before setting user
        if (userDetails) {
          setUser(userDetails);
        } else {
          throw new Error("Invalid user details received");
        }
      })
      .catch(error => {
        setUser(null);
        setToken(null);
        console.error("Login error:", error);
        // Rethrow the error to propagate it to the calling code
        throw error;
      });
  };

  /**
   * Logout function that handles the user logout process.
   *
   * @return {undefined} No return value.
   */
  const logout = token => {
    return authService
      .logout(token)
      .then(response => {
        setUser(null);
        setToken(null);
        return response;
      })
      .catch(error => {
        setUser(null);
        setToken(null);
        console.error("Logout error:", error);
        throw error;
      });
  };

  /**
   * Function to handle user registration.
   * @param {string} email - The user's email.
   * @param {string} name - The user's name.
   * @param {string} username - The user's username.
   * @param {string} password - The user's password.
   * @param {string} password_confirmation - The user's password confirmation.
   */
  const register = (email, name, username, password, password_confirmation) => {
    return authService
      .register(email, name, username, password, password_confirmation)
      .then(response => {
        return response;
      })
      .catch(error => {
        console.error("Registration error:", error);
        throw error;
      });
  };

  /**
   * Retrieves the authenticated user details using the provided token.
   *
   * @param {string} token - The authentication token.
   * @return {Promise} A promise that resolves with the user details or rejects with an error.
   */
  const getAuthUser = token => {
    return authService
      .getAuthUser(token)
      .then(userDetails => {
        // Check if the user data has changed before updating the context
        if (!isEqual(userDetails, user)) {
          setUser(userDetails);
        }
      })
      .catch(error => {
        setUser(null);
        console.error("Error fetching authenticated user:", error);
        throw error;
      });
  };

  // Return the child components wrapped in AuthContext.Provider
  return <AuthContext.Provider value={{ user, token, login, logout, register, getAuthUser }}>{children}</AuthContext.Provider>;
};

/**
 * Returns the authentication context.
 *
 * @return {AuthContext} The authentication context.
 */
export const useAuth = () => {
  const context = useContext(AuthContext);

  if (!context) {
    throw new Error("useAuth must be used within an AuthProvider.");
  }

  return context;
};
