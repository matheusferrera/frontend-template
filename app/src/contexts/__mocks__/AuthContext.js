/* eslint-disable no-undef */

import React from 'react';

import { loginMockData, registerMockData, userMockData } from "../../../test/mockData";


export const AuthContext = React.createContext({
  user: jest.fn().mockResolvedValue(userMockData),
  token: null,
  login: jest.fn().mockResolvedValue(loginMockData),
  logout: jest.fn().mockResolvedValue(null),
  register: jest.fn().mockResolvedValue(registerMockData),
  getAuthUser: jest.fn().mockResolvedValue(userMockData),
});

/**
 * AuthProvider component that handles user authentication and authorization.
 * @param {object} props - React component props.
 * @param {React.ReactNode} props.children - The child components to render.
 * @returns {React.ReactNode} - The rendered child components wrapped in AuthContext.Provider.
 */
// eslint-disable-next-line react/prop-types
export const AuthProvider = ({ children }) => {
  return <AuthContext.Provider value={{ user: userMockData, token: null }}>{children}</AuthContext.Provider>;
};

export const useAuth = () => {
  return React.useContext(AuthContext);
};
