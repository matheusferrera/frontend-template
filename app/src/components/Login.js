import React, { useState, useRef } from "react";
import { useNavigate } from "react-router-dom";
import Form from "react-validation/build/form";
import Input from "react-validation/build/input";
import CheckButton from "react-validation/build/button";
import { useAuth } from "../contexts/AuthContext";

const required = value => {
  if (!value) {
    return (
      <div
        className="mt-1 alert alert-danger"
        role="alert"
      >
        This field is required!
      </div>
    );
  }
};

const Login = () => {
  const navigate = useNavigate();

  const form = useRef();
  const checkBtn = useRef();

  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [loading, setLoading] = useState(false);
  const [message, setMessage] = useState("");

  const { login } = useAuth();

  const onChangeEmail = e => {
    const email = e.target.value;
    setEmail(email);
  };

  const onChangePassword = e => {
    const password = e.target.value;
    setPassword(password);
  };

  const handleLogin = e => {
    e.preventDefault();

    setMessage("");
    setLoading(true);

    form.current.validateAll();

    if (checkBtn.current.context._errors.length === 0) {
      login(email, password)
        .then(() => {
          setLoading(false);
          setMessage("");
          navigate("/profile");
          window.location.reload();
        })
        .catch(error => {
          const resMessage = (error.response && error.response.data && error.response.data.message) || error.message || error.toString();
          setLoading(false);
          setMessage(resMessage);
        });
    } else {
      setLoading(false);
    }
  };

  return (
    <div className="col-md-12">
      <div className="card card-container">
        <img
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
          alt="profile-img"
          className="profile-img-card"
        />

        <Form
          onSubmit={handleLogin}
          ref={form}
        >
          <div className="mb-3">
            <label
              htmlFor="email"
              className="form-label"
            >
              email
            </label>
            <Input
              type="email"
              className="form-control"
              name="email"
              value={email}
              onChange={onChangeEmail}
              validations={[required]}
            />
          </div>

          <div className="mb-3">
            <label
              htmlFor="password"
              className="form-label"
            >
              Password
            </label>
            <Input
              type="password"
              className="form-control"
              name="password"
              value={password}
              onChange={onChangePassword}
              validations={[required]}
            />
          </div>

          <div className="mb-3">
            <button
              className="btn btn-primary btn-block"
              disabled={loading}
            >
              {loading && <span className="spinner-border spinner-border-sm"></span>}
              <span>Login</span>
            </button>
          </div>

          {message && (
            <div className="mb-3">
              <div
                className="alert alert-danger"
                role="alert"
              >
                {message}
              </div>
            </div>
          )}
          <CheckButton
            style={{ display: "none" }}
            ref={checkBtn}
          />
        </Form>
      </div>
    </div>
  );
};

export default Login;
