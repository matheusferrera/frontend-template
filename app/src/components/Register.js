import React, { useState, useRef } from "react";
import Form from "react-validation/build/form";
import Input from "react-validation/build/input";
import CheckButton from "react-validation/build/button";
import { isEmail } from "validator";

import { useAuth } from "../contexts/AuthContext";

const required = value => {
  if (!value) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        This field is required!
      </div>
    );
  }
};

const validEmail = value => {
  if (!isEmail(value)) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        This is not a valid email.
      </div>
    );
  }
};

const vusername = value => {
  if (value.length < 3 || value.length > 20) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        The username must be between 3 and 20 characters.
      </div>
    );
  }
};

const vpassword = value => {
  if (value.length < 6 || value.length > 40) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        The password must be between 6 and 40 characters.
      </div>
    );
  }
};

const Register = () => {
  const form = useRef();
  const checkBtn = useRef();

  const [email, setEmail] = useState("");
  const [name, setName] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [passwordConfirmation, setPasswordConfirmation] = useState("");
  const [successful, setSuccessful] = useState(false);
  const [loading, setLoading] = useState(false);
  const [message, setMessage] = useState("");

  const { register } = useAuth();

  const onChangeEmail = e => {
    const email = e.target.value;
    setEmail(email);
  };

  const onChangeName = e => {
    const name = e.target.value;
    setName(name);
  };

  const onChangeUsername = e => {
    const username = e.target.value;
    setUsername(username);
  };

  const onChangePassword = e => {
    const password = e.target.value;
    setPassword(password);
  };

  const onChangePasswordConfirmation = e => {
    const passwordConfirmation = e.target.value;
    setPasswordConfirmation(passwordConfirmation);
  };

  const handleRegister = async e => {
    e.preventDefault();

    setMessage("");
    setLoading(true);
    setSuccessful(false);

    form.current.validateAll();

    if (password !== passwordConfirmation) {
      setMessage("Password confirmation does not match the password.");
      setSuccessful(false);
      setLoading(false);
      return;
    }

    register(email, name, username, password, passwordConfirmation)
      .then(response => {
        setMessage(response);
        setSuccessful(true);
        setLoading(false);
      })
      .catch(error => {
        const resMessage = (error.response && error.response.data && error.response.data.message) || error.message || error.toString();
        setMessage(resMessage);
        setSuccessful(false);
        setLoading(false);
      });
  };

  return (
    <div className="col-md-12">
      <div className="card card-container">
        <img
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
          alt="profile-img"
          className="profile-img-card"
        />

        <Form
          onSubmit={handleRegister}
          ref={form}
        >
          {!successful && (
            <div>
              <div className="mb-3">
                <label
                  htmlFor="email"
                  className="form-label"
                >
                  Email
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="email"
                  value={email}
                  onChange={onChangeEmail}
                  validations={[required, validEmail]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="name"
                  className="form-label"
                >
                  Name
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="name"
                  value={name}
                  onChange={onChangeName}
                  validations={[required]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="username"
                  className="form-label"
                >
                  Username
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="username"
                  value={username}
                  onChange={onChangeUsername}
                  validations={[required, vusername]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="password"
                  className="form-label"
                >
                  Password
                </label>
                <Input
                  type="password"
                  className="form-control"
                  name="password"
                  value={password}
                  onChange={onChangePassword}
                  validations={[required, vpassword]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="passwordConfirmation"
                  className="form-label"
                >
                  Password Confirmation
                </label>
                <Input
                  type="password"
                  className="form-control"
                  name="passwordConfirmation"
                  value={passwordConfirmation}
                  onChange={onChangePasswordConfirmation}
                  validations={[required, vpassword]}
                />
              </div>

              <div className="mb-3">
                <button
                  className="btn btn-primary btn-block"
                  disabled={loading}
                >
                  {loading && <span className="spinner-border spinner-border-sm"></span>}
                  <span>Sign Up </span>
                </button>
              </div>
            </div>
          )}

          {message && (
            <div className="mb-3">
              <div
                className={successful ? "alert alert-success" : "my-1 alert alert-danger"}
                role="alert"
              >
                {message}
              </div>
            </div>
          )}
          <CheckButton
            style={{ display: "none" }}
            ref={checkBtn}
          />
        </Form>
      </div>
    </div>
  );
};

export default Register;
