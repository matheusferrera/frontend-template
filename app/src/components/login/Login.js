import React from "react";
import { useState } from "react";
import { useNavigate } from "react-router-dom";

import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";
import PropTypes from "prop-types";

import { useAuth } from "../../contexts/AuthContext";
import AvisoDePrivacidadeModal from "../modals/AvisoDePrivacidadeModal";
import TermoDeUsoModal from "../modals/TermoDeUsoModal";
import LoginForm from "./LoginForm";

function backendUserType(userRota) {
  if (userRota === "/admin") return "admin";
  if (userRota === "/cidadao") return "trabalhador";
  return "parceiro";
}

const Login = ({ userRota = "" }) => {
  const { login } = useAuth();

  const navigate = useNavigate();

  const [loading, setLoading] = useState(false);

  const recaptchaRef = React.createRef();

  const [termoDeUsoModal, setTermoDeUsoModal] = useState(false);
  const [avisoDePrivacidadeModal, setAvisoDePrivacidadeModal] = useState(false);

  const handleTermoDeUsoShow = () => setTermoDeUsoModal(true);
  const handleTermoDeUsoClose = () => setTermoDeUsoModal(false);

  const handleAvisoDePrivacidadeShow = () => setAvisoDePrivacidadeModal(true);
  const handleAvisoDePrivacidadeClose = () => setAvisoDePrivacidadeModal(false);

  const handleSubmit = ({ username, password }, { setSubmitting }) => {
    const token = recaptchaRef.current.getValue();

    if (!token) {
      alert("Por favor, confirme que você não é um robô");
      setSubmitting(false);
      return;
    }

    const promise = login(username, password, backendUserType(userRota));

    setLoading(true);

    promise
      .then(() => {
        navigate("/");
      })
      .catch(error => alert(error.response?.data?.message || error.message || error.toString()))
      .finally(() => {
        setLoading(false);
        setSubmitting(false);
      });
  };

  return (
    <Box
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        minHeight: "100vh",
      }}
    >
      <Stack
        alignItems="center"
        justifyContent="center"
        sx={{ height: 1 }}
      >
        <Card
          sx={{
            p: 5,
            width: 1,
            maxWidth: 625,
            height: 1,
            maxHeight: 700,
          }}
        >
          <Typography variant="h3">Bem-vindo(a)</Typography>

          <Typography
            variant="body2"
            sx={{ mt: 2, mb: 3, color: "text.grey" }}
          >
            Acesse aqui sua conta Progredir
          </Typography>

          <LoginForm
            loading={loading}
            handleSubmit={handleSubmit}
            recaptchaRef={recaptchaRef}
            handleAvisoDePrivacidadeShow={handleAvisoDePrivacidadeShow}
            handleTermoDeUsoShow={handleTermoDeUsoShow}
          />

          {userRota == "/parceiro" && (
            <>
              <Button
                variant="outlined"
                sx={{ mt: 2 }}
                href="/register"
              >
                Criar conta
              </Button>

              <Button
                variant="outlined"
                sx={{ mt: 2, ml: 2 }}
                href="/activate"
              >
                Ativar conta
              </Button>
            </>
          )}

          <TermoDeUsoModal
            showModal={termoDeUsoModal}
            handleClose={handleTermoDeUsoClose}
          />
          <AvisoDePrivacidadeModal
            showModal={avisoDePrivacidadeModal}
            handleClose={handleAvisoDePrivacidadeClose}
          />
        </Card>
      </Stack>
    </Box>
  );
};

Login.propTypes = {
  userRota: PropTypes.string,
};

export default Login;
