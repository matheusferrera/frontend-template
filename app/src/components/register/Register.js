import React from "react";
import { useState } from "react";
import { useNavigate } from "react-router-dom";

import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";

import { useAuth } from "../../contexts/AuthContext";
import AvisoDePrivacidadeModal from "../modals/AvisoDePrivacidadeModal";
import TermoDeUsoModal from "../modals/TermoDeUsoModal";
import RegisterForm from "./RegisterForm";

const Register = () => {
  const { register } = useAuth();
  const navigate = useNavigate();

  const [loading, setLoading] = useState(false);

  const recaptchaRef = React.createRef();

  const [termoDeUsoModal, setTermoDeUsoModal] = useState(false);
  const [avisoDePrivacidadeModal, setAvisoDePrivacidadeModal] = useState(false);

  const handleTermoDeUsoShow = () => setTermoDeUsoModal(true);
  const handleTermoDeUsoClose = () => setTermoDeUsoModal(false);

  const handleAvisoDePrivacidadeShow = () => setAvisoDePrivacidadeModal(true);
  const handleAvisoDePrivacidadeClose = () => setAvisoDePrivacidadeModal(false);

  const handleSubmit = (values, { setSubmitting }) => {
    const token = recaptchaRef.current.getValue();

    setLoading(true);

    if (!token) {
      alert("Por favor, confirme que você não é um robô");
      setLoading(false);
      setSubmitting(false);
      return;
    }

    register(values.email, values.name, values.password, values.passwordConfirmation)
      .then(response => {
        navigate("/login");
        alert(response);
      })
      .catch(error => {
        const resMessage = (error.response && error.response.data && error.response.data.message) || error.message || error.toString();
        alert(resMessage);
      })
      .finally(() => {
        setLoading(false);
        setSubmitting(false);
      });
  };

  return (
    <Box
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        minHeight: "100vh",
      }}
    >
      <Stack
        alignItems="center"
        justifyContent="center"
        sx={{ height: 1 }}
      >
        <Card
          sx={{
            p: 5,
            width: 1,
            maxWidth: 625,
            height: 1,
            maxHeight: 900,
          }}
        >
          <Typography variant="h3">Registrar-se</Typography>

          <Typography
            variant="body2"
            sx={{ mt: 2, mb: 3, color: "text.grey" }}
          >
            Crie o seu perfil Progredir
          </Typography>

          <RegisterForm
            loading={loading}
            handleSubmit={handleSubmit}
            recaptchaRef={recaptchaRef}
            handleAvisoDePrivacidadeShow={handleAvisoDePrivacidadeShow}
            handleTermoDeUsoShow={handleTermoDeUsoShow}
          />

          <Button
            variant="outlined"
            sx={{ mt: 2 }}
            href="/parceiro_login"
          >
            Já possuo conta
          </Button>

          <Button
            variant="outlined"
            sx={{ mt: 2, ml: 2 }}
            href="/activate"
          >
            Ativar conta
          </Button>

          <TermoDeUsoModal
            showModal={termoDeUsoModal}
            handleClose={handleTermoDeUsoClose}
          />
          <AvisoDePrivacidadeModal
            showModal={avisoDePrivacidadeModal}
            handleClose={handleAvisoDePrivacidadeClose}
          />
        </Card>
      </Stack>
    </Box>
  );
};

export default Register;
