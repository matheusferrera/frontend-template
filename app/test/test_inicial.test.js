import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import { waitFor } from "@testing-library/react";
import axios from "axios";

import Home from "../src/components/Home";
import Login from "../src/components/Login";
import NotFound from "../src/components/NotFound";
import Register from "../src/components/Register";
import { AuthProvider } from "../src/contexts/AuthContext";
import customAxios from "./__mocks__/axios";

jest.mock("axios"); // Mockando o módulo axios

describe("Teste de componentes", () => {
  it("Home renderizada corretamente", () => {
    const component = renderer.create(<Home />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Not found renderizada corretamente", () => {
    const component = renderer.create(<NotFound />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Register renderizada corretamente", () => {
    const component = renderer.create(
      <Router>
        <AuthProvider>
          <Register />
        </AuthProvider>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Login renderizada corretamente", () => {
    const component = renderer.create(
      <Router>
        <AuthProvider>
          <Login />
        </AuthProvider>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Teste de login usando mock", async () => {
    renderer.create(
      <Router>
        <AuthProvider>
          <Login />
        </AuthProvider>
      </Router>,
    );

    // Utiliza a função `Promise.all` para aguardar a conclusão de múltiplas promessas
    await Promise.all([
      // Utiliza o ato assíncrono para aguardar a conclusão de uma ação assíncrona
      act(async () => {
        await axios.get("/login"); // Simula uma chamada à API usando a função `get` do Axios
        waitFor(() => expect(customAxios.get).toHaveBeenCalledTimes(1)); // Aguarda até que a função `get` do Axios tenha sido chamada uma vez (waitFor)
      }),
    ]);

    // Verifica se a função `get` do Axios foi chamada uma vez durante o teste
    expect(axios.get).toHaveBeenCalledTimes(1);
  });
});
