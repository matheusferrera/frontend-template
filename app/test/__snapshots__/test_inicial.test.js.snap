// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de componentes Home renderizada corretamente 1`] = `
<div
  className="container"
>
  <header
    className="jumbotron"
  >
    <h3>
      Página em construção
    </h3>
  </header>
</div>
`;

exports[`Teste de componentes Login renderizada corretamente 1`] = `
<div
  className="col-md-12"
>
  <div
    className="card card-container"
  >
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="text-input"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="password"
        >
          Senha
        </label>
        <input
          className="text-input"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Senha"
          type="password"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <button
          disabled={false}
          type="submit"
        >
          <span>
            Login
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Teste de componentes Not found renderizada corretamente 1`] = `
<div>
  <h1>
    404 - Página não encontrada
  </h1>
  <p>
    Ops! A página que você está procurando não existe.
  </p>
</div>
`;

exports[`Teste de componentes Register renderizada corretamente 1`] = `
<div
  className="col-md-12"
>
  <div
    className="card card-container"
  >
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="text-input"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="name"
          >
            Nome
          </label>
          <input
            className="text-input"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Nome"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="username"
          >
            Nome de usuário
          </label>
          <input
            className="text-input"
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Nome de usuário"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="password"
          >
            Senha
          </label>
          <input
            className="text-input"
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Senha"
            type="password"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="passwordConfirmation"
          >
            Confirmação de senha
          </label>
          <input
            className="text-input"
            id="passwordConfirmation"
            name="passwordConfirmation"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Confirmação de senha"
            type="password"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <button
            disabled={false}
            type="submit"
          >
            <span>
              Registrar
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
